trestle/core/commands/author/profile.py

```diff
@@ -336,47 +348,38 @@ def assemble_profile(
                     for part in add.parts:
                         if part.name not in allowed_sections:
                             raise TrestleError(f'Profile has alter with name {part.name} not in allowed sections.')
-        ProfileAssemble._replace_alter_adds(new_profile, found_alters)
+        ProfileAssemble._replace_alter_adds(parent_prof, found_alters)
         if set_parameters:
-            ProfileAssemble._replace_modify_set_params(new_profile, param_dict, param_map)
-
-        new_profile_dir = trestle_root / f'profiles/{new_profile_name}'
-        new_profile_path = new_profile_dir / 'profile.json'
+            ProfileAssemble._replace_modify_set_params(parent_prof, param_dict, param_map)
 
         if version:
-            new_profile.metadata.version = com.Version(__root__=version)
-
-        if new_profile_path.exists():
-            # first do simple and fast tests of changes so that the full equality check is only done as last resort
-            _, _, existing_profile = ModelUtils.load_distributed(new_profile_path, trestle_root)
-            # need special handling in case modify is None in either or both profiles
-            no_changes = not existing_profile.modify and not new_profile.modify
-            both_modifies_present = existing_profile.modify and new_profile.modify
-            if both_modifies_present:
-                no_changes = (
-                    existing_profile.modify.alters == new_profile.modify.alters
-                    and existing_profile.modify.set_parameters == new_profile.modify.set_parameters
-                )
-            if no_changes and existing_profile.metadata.version != new_profile.metadata.version:
-                no_changes = False
-            # at this point if no known changes then do the full and possibly expensive comparison to confirm
-            if no_changes and existing_profile == new_profile:
+            parent_prof.metadata.version = com.Version(__root__=version)
+
+        assem_prof_path = ModelUtils.path_for_top_level_model(
+            trestle_root, assem_prof_name, prof.Profile, new_content_type
+        )
+
+        if assem_prof_path.exists():
+            _, _, existing_prof = ModelUtils.load_distributed(assem_prof_path, trestle_root)
+            if ModelUtils.models_are_equivalent(existing_prof, parent_prof):
                 logger.info('Assembled profile has no changes so no update of existing file.')
                 return CmdReturnCodes.SUCCESS.value
 
         if regenerate:
-            new_profile, _, _ = ModelUtils.regenerate_uuids(new_profile)
-        ModelUtils.update_last_modified(new_profile)
+            parent_prof, _, _ = ModelUtils.regenerate_uuids(parent_prof)
+        ModelUtils.update_last_modified(parent_prof)
 
-        if new_profile_dir.exists():
+        if assem_prof_path.parent.exists():
```

Comment:
Will fail if assem_prof_path doesn't exist

Reply:
I think it's ok - existence isn't required to know parent.  It's done by operating on the path as a string.
Example:
In empty but existing directory `/tmp/farble`:
```
>>> import pathlib
>>> p = pathlib.Path('/tmp/farble/foo/bar/cow.txt')
>>> p
WindowsPath('/tmp/farble/foo/bar/cow.txt')
>>> p.exists()
False
>>> p.parent.exists()
False
>>> p.parent
WindowsPath('/tmp/farble/foo/bar')
>>> p.parent.mkdir(parents=True)
>>> p
WindowsPath('/tmp/farble/for/bar/cow.txt')
```